lib/screens/profile/profile_repository.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:math';
+
+import 'package:ecellapp/core/res/errors.dart';
+import 'package:flutter/material.dart';
+
+@immutable
+abstract class ProfileRepository {
+  /// Takes in `token` , gives the user details and throws a suitable exception if something goes wrong.
+  Future<Map<String, dynamic>> profile(String token);
+}
+
+class FakeProfileRepository implements ProfileRepository {
+  @override
+  Future<Map<String, dynamic>> profile(String token) async {
+    // Simulate network delay
+    await Future.delayed(Duration(seconds: 1));
+
+    if (Random().nextBool()) {
+      // random network error
+      throw NetworkException();
+    } else {
+      // fake successful login (the data entered here same as in the API Doc example)
+      return {
```

Comment:
instead of returning a map, it should return a user object. create a user model class, create an instance of that here and return that. Also, skip unimportant fields like `applied`, `bquiz_score`, `verified`, `facebook`, `user_type`, `linkedin`, `id`

Reply:
done!